addons/portal/models/res_partner.py

```diff
@@ -18,3 +35,8 @@ def can_edit_vat(self):
         edit it (as in backend)."""
         self.ensure_one()
         return not self.parent_id
+
+    def can_edit_company(self):
+        """ `company_name` can be edited if the vat number is not set and set partner company."""
+        self.ensure_one()
+        return self.vat
```

Comment:
Nothing is correct with this method, the name, the comment that says the opposite of what the method is doing, etc... it's the same for changes in `/account`.